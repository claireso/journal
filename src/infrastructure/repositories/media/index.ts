export { default as MediaRepositoryImpl } from './MediaRepositoryImpl'
