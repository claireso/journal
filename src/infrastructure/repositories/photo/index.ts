export { default as PhotoRepositoryImpl } from './PhotoRepositoryImpl'
