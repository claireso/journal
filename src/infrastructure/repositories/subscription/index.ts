export { default as SubscriptionRepositoryImpl } from './SubscriptionRepositoryImpl'
