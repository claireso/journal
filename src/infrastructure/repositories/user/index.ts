export { default as UserRepositoryImpl } from './UserRepositoryImpl'
