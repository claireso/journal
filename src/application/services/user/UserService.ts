import { UserRepository } from '@domain/repositories'

export default class MediaService {
  private repository: UserRepository
  private logger: unknown

  constructor(repository: UserRepository, logger: unknown) {
    this.repository = repository
    this.logger = logger
  }

  async authenticate(username: string, password: string) {
    return this.repository.getByCredentials(username, password)
  }
}
