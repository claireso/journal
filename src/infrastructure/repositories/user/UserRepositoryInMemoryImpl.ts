import { User } from '@domain/entities'
import { UserRepository } from '@domain/repositories'

export default class UserRepositoryInMemoryImpl implements UserRepository {
  private users: User[]

  constructor() {
    this.users = [
      {
        id: 1,
        created_at: new Date('2024-10-05T17:12:40.400Z'),
        updated_at: new Date('2024-10-05T17:12:40.400Z'),
        cid: 'c123',
        username: 'janedoe'
      }
    ]
  }

  async getByCredentials(username: string, password: string): Promise<User | null> {
    const user = this.users.find((user) => user.username === username)
    return user || null
  }
}
