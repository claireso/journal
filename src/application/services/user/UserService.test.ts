import { UserRepository } from '@domain/repositories'
import UserRepositoryInMemoryImpl from '@infrastructure/repositories/user/UserRepositoryInMemoryImpl'
import UserService from './UserService'

describe('application/UserService', () => {
  let userRepository: UserRepository
  let userService: UserService

  beforeEach(() => {
    userRepository = new UserRepositoryInMemoryImpl()
    userService = new UserService(userRepository, console)
  })

  describe('GET', () => {
    it('should get user by credentials', async () => {
      // act
      const user = await userService.authenticate('janedoe', '123')
      // assert
      expect(user).toMatchSnapshot()
    })

    it('should not get user by credential', async () => {
      // act
      const user = await userService.authenticate('janedoe123', '123')
      // assert
      expect(user).toBeNull()
    })
  })
})
