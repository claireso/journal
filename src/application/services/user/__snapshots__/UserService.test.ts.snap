// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`application/UserService GET should get user by credentials 1`] = `
{
  "cid": "c123",
  "created_at": 2024-10-05T17:12:40.400Z,
  "id": 1,
  "updated_at": 2024-10-05T17:12:40.400Z,
  "username": "janedoe",
}
`;
